presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -171,6 +189,11 @@ public OrcSelectiveRecordReader(
         streamReaderOrder = orderStreamReaders(columnTypes.keySet().stream().filter(index -> this.constantValues[index] == null).collect(toImmutableSet()), columnsWithFilters, filterFunctionInputs);
     }
 
+    private static boolean containsNonNullFilter(Map<Subfield, TupleDomainFilter> columnFilters)
+    {
+        return columnFilters != null && !columnFilters.entrySet().stream().map(Map.Entry::getValue).allMatch(filter -> filter.testNull());
```

Comment:
- entrySet().stream().map(Map.Entry::getValue) can be simplified to values().stream()
- use TupleDomainFilter::testNull in allMatch
- should we check if filters are deterministic as well?

return columnFilters != null && !columnFilters.values().stream().allMatch(TupleDomainFilter::testNull);